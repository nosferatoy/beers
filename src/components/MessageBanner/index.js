import React from 'react';
import cx from 'classnames';

import './index.scss';

export default function MessageBanner({ text, type }) {
	return (
		<div className={cx('message', `message__${type}`)}>
			<p className={cx('message--text')}>{text}</p>
		</div>
	);
}
