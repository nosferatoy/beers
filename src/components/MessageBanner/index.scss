.message {
    padding: 16px;
    border-radius: 3px;
    margin-top: 20px;
    margin-bottom: 20px;

    &__danger {
        background-color: var(--color-danger);
        border: 1px solid var(--color-danger-border);
        color: var(--color-danger-text);
      }
    
    &__primary {
        background-color: var(--color-primary);
        border: 1px solid var(--color-primary-border);
        color: var(--color-primary-text);
      }

    &--text {
        margin: 0;
      } 
}