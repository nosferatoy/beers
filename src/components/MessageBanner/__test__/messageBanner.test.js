import React from 'react';
import { render } from '@testing-library/react';
import MessageBanner from '../index';

describe('Test MessageBanner component', () => {
	it('Render MessageBanner Danger', () => {
		const { container } = render(
			<MessageBanner text={'Danger Test'} type="danger" />
		);
		expect(container).toMatchSnapshot();
	});

	it('Render MessageBanner Primary', () => {
		const { container } = render(
			<MessageBanner text={'Primary Text'} type="primary" />
		);
		expect(container).toMatchSnapshot();
	});
});
