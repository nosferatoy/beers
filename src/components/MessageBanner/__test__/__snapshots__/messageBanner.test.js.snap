// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test MessageBanner component Render MessageBanner Danger 1`] = `
<div>
  <div
    class="message message__danger"
  >
    <p
      class="message--text"
    >
      Danger Test
    </p>
  </div>
</div>
`;

exports[`Test MessageBanner component Render MessageBanner Primary 1`] = `
<div>
  <div
    class="message message__primary"
  >
    <p
      class="message--text"
    >
      Primary Text
    </p>
  </div>
</div>
`;
